app/controllers/concerns/base_application_controller_behavior.rb

```diff
@@ -101,9 +102,13 @@ def prepare_exception_notifier
     end
 
     def set_sentry_user
-      return unless ENV['WAREHOUSE_SENTRY_DSN'].present?
+      return unless ENV['WAREHOUSE_SENTRY_DSN'].present? && Sentry.initialized?
+      return unless defined?(current_app_user)
+      return unless current_app_user.is_a?(User) || current_app_user.is_a?(Hmis::User)
 
-      Sentry.configure_scope { |scope| scope.set_user(id: current_app_user.id, email: current_app_user.email) } if Sentry.initialized? && defined?(current_app_user) && current_app_user.is_a?(User)
+      Sentry.configure_scope do |scope|
+        scope.set_user(id: current_app_user.id, email: current_app_user.email)
```

Comment:
Tested on QA:

https://green-river.sentry.io/issues/4739902962/events/latest/?project=4503977346662400&referrer=latest-event

It seems like the email gets filtered out anyway, but the ID can be helpful.